shap/utils/_types.py

```diff
@@ -0,0 +1,9 @@
+from typing import TypeVar, Union
+
+import numpy as np
+import pandas as pd
+import scipy.sparse
+
+# TODO: use TypeAlias (when we drop python 3.9)
+_ArrayLike = Union[np.ndarray, pd.DataFrame, pd.Series, list, scipy.sparse.spmatrix]
+_ArrayT = TypeVar("_ArrayT", np.ndarray, pd.DataFrame, pd.Series, scipy.sparse.spmatrix, list)
```

Comment:
Any reason why we're going for `_ArrayT` instead of `ArrayT`? (If it's for keeping it "private", I'm wondering if that's necessary?)

Reply:
By convention, a leading underscore designates a private object (strictly speaking "non-public"), that is not part of the public API for users. These objects are typically hidden from IDE tool tips and IPython auto complete.

Whilst public interfaces must be tested and protected from sudden breaking changes, private objects are for internal use only and can be refactored more freely.

It's helpful for us as maintainers to minimise the public API and keep any helper functions / variables as private, as it allows us to refactor the internal code more freely.

Reply:
Sure, I appreciate the difference bet private and public API, let's leave it as such since numpy is doing the same.